library/hmac_drbg.c

```diff
@@ -74,40 +77,74 @@ int mbedtls_hmac_drbg_update_ret( mbedtls_hmac_drbg_context *ctx,
         mbedtls_md_get_handle( &ctx->md_ctx ) );
     unsigned char rounds = ( additional != NULL && add_len != 0 ) ? 2 : 1;
     unsigned char sep[1];
+    volatile unsigned char flow_counter = 0;
```

Comment:
Suggestion for improvement: you can save about 20 bytes of code size by making this `unsigned int` instead. Handling chars incurs some overhead (for example, the STRB instruction can't reference SP directly while the STR instruction can).

Reply:
Thanks for detecting this, it is now fixed in the following commit. 
It saves 20 bytes of ROM with minimal effort.